src/routers/alpha-router/gas-models/v3/v3-heuristic-gas-model.ts

```diff
@@ -259,6 +269,81 @@ export class V3HeuristicGasModelFactory extends IOnChainGasModelFactory {
         throw err;
       }
 
+      /*
+        We want to get the gas cost in terms of the quote token.
+        Sometimes, the quote token / native token pair is very illiquid and the price is incorrect.
+        We also have access to the input token / native token pair, which is more likely to be liquid.
+          - if it's not, then it is an illiquid direct swap trade
+        We can get the gas cost in terms of the input token, 
+          using the inputTokenNativePool and quoting for the amount of ETH required for gas
+        Then, we can use the execution price to get the respective amounts of quote tokens
+        Then, we can take the minimum of the initial quoted gas cost in terms of quote token
+          and our calculated one.
+      */
+      try {
```

Comment:
Why surround with a try? I don't see any reason why this should fail

Reply:
Was mirroring the try/catch on line 255:
```

      let gasCostInTermsOfQuoteToken: CurrencyAmount;
      try {
        // native token is base currency
        gasCostInTermsOfQuoteToken = nativeTokenPrice.quote(
          totalGasCostNativeCurrency
        ) as CurrencyAmount;
      } catch (err) {
        log.info(
          {
            nativeTokenPriceBase: nativeTokenPrice.baseCurrency,
            nativeTokenPriceQuote: nativeTokenPrice.quoteCurrency,
            gasCostInEth: totalGasCostNativeCurrency.currency,
          },
          'Debug eth price token issue'
        );
        throw err;
      }
```

Reply:
yeah not sure its needed, unless `<Price>.quote()` can fail